CorrelatorSharp.Logging/LogManager.cs

```diff
@@ -14,5 +17,30 @@ public static ILogger GetLogger(string name)
             return LoggingConfiguration.LogManager.GetLogger(name);
         }
 
+        // From NLog's LogManager code
+        private static string GetClassFullName()
+        {
+            string className;
+            Type declaringType;
+            var framesToSkip = 2;
+
+            do
+            {
+                var frame = new StackFrame(framesToSkip, false);
+                var method = frame.GetMethod();
+                declaringType = method.DeclaringType;
+                if (declaringType == null)
+                {
+                    className = method.Name;
+                    break;
+                }
+
+                framesToSkip++;
+                className = declaringType.FullName;
+            } while (declaringType.Module.Name.Equals("mscorlib.dll", StringComparison.OrdinalIgnoreCase));

```

Comment:
Can you look into a way to not make this dependent on 'mscorlib.dll', which doesn't exist in portable or .Net core? maybe you get do typeof(String).assembly/module or similar?